### PR TITLE
[lang] Remove redundant codegen of integer pow

### PR DESCRIPTION
**Pull request by strongoier**
Related issue = #5915

As `pow()` with integer exponent is already demoted (#6044), there's no need for backends to handle integer pows.

### CODE REVIEW

File: taichi/codegen/cuda/codegen_cuda.cpp [hunk 1]
```diff
@@ -667,10 +667,6 @@ class TaskCodeGenCUDA : public TaskCodeGenLLVM {
         llvm_val[stmt] = create_call("__nv_powf", {lhs, rhs});
       } else if (ret_type->is_primitive(PrimitiveTypeID::f64)) {
         llvm_val[stmt] = create_call("__nv_pow", {lhs, rhs});
-      } else if (ret_type->is_primitive(PrimitiveTypeID::i32)) {
```
Comment (ailzhang) [inline taichi/codegen/cuda/codegen_cuda.cpp:670]: nit: can we add a comment here to redirect readers to the demote pass? :P I'm afraid people might accidentally miss that and add these back in the future lol

**Review by ailzhang — APPROVED** [review]
Thanks!

File: taichi/codegen/cuda/codegen_cuda.cpp (hunk 1 above)
Comment (strongoier) [reply taichi/codegen/cuda/codegen_cuda.cpp:670]: SG!

**Review by AD1024 — APPROVED** [review]
LGTM!

**Pull request merged**